azure/services/scalesets/scalesets.go

```diff
@@ -637,6 +637,14 @@ func getSecurityProfile(vmssSpec azure.ScaleSetSpec, sku resourceskus.SKU) (*com
 
 // base64EncodedHash transforms a VMSS into json and then creates a sha256 hash of the data encoded as a base64 encoded string
 func base64EncodedHash(vmss compute.VirtualMachineScaleSet) (string, error) {
+	// Setting Admin Password is not supported but an initial password is required for Windows
+	// Don't include it in the hash since it is generated and won't be the same each the spec is created (#1182)
+	tmpPass := vmss.VirtualMachineProfile.OsProfile.AdminPassword
+	vmss.VirtualMachineProfile.OsProfile.AdminPassword = nil
+	defer func() {
+		vmss.VirtualMachineProfile.OsProfile.AdminPassword = tmpPass
```

Comment:
do we really need to set this value again here? the function param isn't a pointer so it shouldn't modify the actual value of the vmss outside the function no?

Reply:
FYI: Hash calculation goes away in MachinePoolMachines. The reasoning is that we need to do a get and a diff each time. There is little gain in having a hash if you can just compare actual state. The downside is that we have to be able to transform Azure state to CAPZ Model state (compute.VMSS -> azure.VMSS), which we don't aways do for resources.

I might be nice to be able to have converters or enforce conversion from Azure model to CAPZ model.

Reply:
I think this might be needed. `compute.VirtualMachineScaleSet` contains an embedded ptr to `*VirtualmachineScaleSetProperties`, so the ptr value is copied and points to the same memory even though it looks like the param should not modify the caller.